src/binding.jl

```diff
@@ -482,6 +510,28 @@ function push_pattern!(patterns::Vector{BoundPattern}, binder::BinderContext, pa
     get_temp(binder, pat)
 end
 
+function parse_kw_param(param, source)
+    if is_expr(param, :kw, 2)
+        # (; x = p)
+        field_name = param.args[1]
+        # bind both the pattern p, but not the field name.
+        pattern_source = param.args[2]
+    elseif is_expr(param, :(::), 2) && param.args[1] isa Symbol
+        # (; x::T) -- equivalent to (; x=(x::T))
+        field_name = param.args[1]
+        # bind both the field name and pattern `::T`
+        pattern_source = param
+    elseif param isa Symbol
+        # (; x) -- equivalent to (; x=x)
+        field_name = param
+        pattern_source = param
+    else
+        error("$(location.file):$(location.line): Unexpected named parameter " *
```

Comment:
`location` is not defined here. Please test this.

Reply:
done